config/reader.go

```diff
@@ -390,24 +390,34 @@ func processConfigLocalConfig(unprocessedConfig *Config, logger logging.Logger)
 	return processConfig(unprocessedConfig, false, logger)
 }
 
+// processConfig processes the config passed in. The config can be either JSON or gRPC derived.
 func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Logger) (*Config, error) {
+	// Ensure validates the config but also substitutes in some defaults. Implicit dependencies for builtin resource
+	// models are not filled in until attributes are converted.
 	if err := unprocessedConfig.Ensure(fromCloud, logger); err != nil {
 		return nil, err
 	}
 
+	// we cached the unprocessed config, so make a copy before changing it too much. Also ensures validation
+	// happens again on resources, remotes, and modules.
 	cfg, err := unprocessedConfig.CopyOnlyPublicFields()
 	if err != nil {
 		return nil, errors.Wrap(err, "error copying config")
 	}
 
+	// Copy does not preserve ConfigFilePath and we need to pass it along manually
```

Comment:
Why is that again? This looks like an exported field?

Reply:
ConfigFilePath is not a JSON-exported field tho - updated comment